packages/fxa-shared/db/redis.ts

```diff
@@ -188,83 +192,138 @@ export class RedisShared {
+    return result;
   }
 
   async getRefreshTokens(uid: Buffer | string) {
+    this.metrics?.increment('redis.getRefreshTokens');
+    const span = tracer.startSpan('redis.getRefreshTokens');
     try {
       const p1 = this.redis.hgetall(hex(uid));
       const p2 = this.resolveInMs(p1, this.timeoutMs, {});
       const tokens = await Promise.race([p1, p2]);
+      span.setAttribute(
+        'redis.getRefreshTokens.tokens.length',
+        Object.keys(tokens).length
+      );
+      this.metrics?.histogram(
+        'redis.getRefreshTokens.tokens.length',
+        Object.keys(tokens).length
+      );
       for (const id of Object.keys(tokens)) {
         tokens[id] = RefreshTokenMetadata.parse(tokens[id]);
       }
       return tokens;
     } catch (e) {
+      this.metrics?.increment('redis.getRefreshTokens.error');
       this.log?.error('RedisShared', { error: e });
       return {};
+    } finally {
+      span.end();
     }
   }
 
   async pruneSessionTokens(uid: string, tokenIds: string[] = []) {
+    this.metrics?.increment('redis.pruneSessionTokens');
+    const span = tracer.startSpan('redis.pruneSessionTokens');
     const p1 = this.redis.pruneSessionTokens(uid, JSON.stringify(tokenIds));
     const p2 = this.rejectInMs(p1, this.timeoutMs);
-    return Promise.race([p1, p2]);
+    const result = Promise.race([p1, p2]);
+    span.end();
+    return result;
   }
 
   async pruneRefreshTokens(
     uid: Buffer | String,
     tokenIdsToPrune: Buffer[] | string[]
   ) {
+    this.metrics?.increment('redis.pruneRefreshTokens');
+    const span = tracer.startSpan('redis.pruneRefreshTokens');
     const p1 = this.redis.hdel(hex(uid), ...tokenIdsToPrune.map((v) => hex(v)));
     const p2 = this.resolveInMs(p1, this.timeoutMs);
-    return await Promise.race([p1, p2]);
+    const result = await Promise.race([p1, p2]);
+    span.end();
+    return result;
   }
 
   async getSessionTokens(uid: string) {
+    this.metrics?.increment('redis.getSessionTokens');
+    const span = tracer.startSpan('redis.getSessionTokens');
     try {
       const p1 = this.redis.getSessionTokens(uid);
       const p2 = this.rejectInMs(p1, this.timeoutMs);
       const value = await Promise.race([p1, p2]);
+
+      if (value?.length > 0) {
+        span.setAttribute('tokens.length', value.length);
+        this.metrics?.histogram(
+          'redis.getSessionTokens.tokens.length',
+          value.length
+        );
+      }
       return JSON.parse(value as string);
     } catch (e) {
       this.log?.error('RedisShared', {
         error: e,
       });
       return {};
+    } finally {
+      span.end();
     }
   }
 
   async getAccessTokens(uid: Buffer | String) {
+    this.metrics?.increment('redis.getAccessTokens');
+    const span = tracer.startSpan('redis.getAccessTokens');
     try {
       const values = await this.redis.getAccessTokens(hex(uid));
+
+      if (values?.length) {
+        span.setAttribute('tokens.length', values.length);
```

Comment:
I guess similar Q as above, should this be `'tokens.length'` or `'redis.getAccessTokens.tokens.length'`?